docs/TOKENS.md

```diff
@@ -13,133 +13,141 @@ Spoilers. My love for Astro has grown deeper.
+Tokens are a form of authentication that verifies whether the user can access parts of an application. In this case, we want them to authenticate the user and their privileges within a video call. 
 
-The token is generated based on the HMAC-SHA256 cryptographic algorithm. We don't need to dive into cybersecurity theory; just know that it's an industry standard for authentication and it will give you a code that verifies your user's identity. In Node we can generate this token using the agora-token package.
+The token is generated based on the HMAC-SHA256 cryptographic algorithm. We don't need to dive into cybersecurity theory; just know that it's an industry standard for authentication, and it will give you a code that verifies your user's identity. We can generate this token in Node using the `agora-token` package.
 
 ## Setting up an Endpoint in Astro
-To create an endpoint in Astro, add a Javascript or Typescript file to the `pages` directory. The name of the file should have the datatype extension in it. So your file name will look like `<name>.<return-type>.ts`. The Javascript and Typescript extensions will be removed during the build process.
+To create an endpoint in Astro, add a Javascript or Typescript file to the `pages` directory. The name of the file should have the datatype extension. So your file name will look like `<name>.<return-type>.ts`. The Javascript and Typescript extensions will be removed during the build process.
 
-The file path we will be using for generating our token is `pages/rtc/[channel]/[role]/[uid].json.ts`. This endpoint will return JSON data, so the file name ends with `.json.ts`. Our endpoint will also need parameters. In Astro, you use brackets for the file/folder name. Since the Typescript extension gets removed, and the `pages` directory doesn't show up in the URL, our final endpoint will be: `rtc/{channel}/{role}/{uid}.json`.
+The file path we will use to generate our token is `pages/api/token.json.ts`. This endpoint will return JSON data, so the file name ends with `.json.ts`. I like to put all endpoints within an `api` folder so that they are separated from front-end routes. Since the Typescript extension gets removed and the `pages` directory doesn't show up in the URL, our final endpoint will be `api/token.json`.
 
 ## Define the endpoint
-Our endpoint will be a `GET` request because our objective is to retrieve a token. You can define this in Astro by exporting a `GET` function and returning a JSON Response.
-
-We also import the `agora-token` package and bring in our `APP_ID` and `APP_CERTIFICATE`, which we defined in the `.env` file.
-
+Because our endpoint will need some input information, we will use a `POST` request and retrieve the information from the body. You can define this in Astro by exporting a `POST` function and returning a Response object.
```

Comment:
The word retrieve doesn't make sense in this context. The users needs to pass data and you are correctly saying to use POST, but the body is the way of passing the information from the client to the server . 

>  we will use a `POST` request so we can pass the input as the request body.

Reply:
updated to your suggestion